internal/report/retrieval.go

```diff
@@ -0,0 +1,51 @@
+package report
+
+import (
+	"encoding/xml"
+	"github.com/jenkins-x-apps/jx-app-jacoco/internal/util"
+	"github.com/jenkins-x/jx/pkg/cloud/buckets"
+	"github.com/jenkins-x/jx/pkg/jx/cmd"
+	"time"
+)
+
+var (
+	timeout = time.Second*30
+	retriever Retriever = &defaultRetriever{}
+)
+
+type Retriever interface {
+	getRawReport(url string) ([]byte, error)
+}
+
+type defaultRetriever struct {
+}
+
+func (r *defaultRetriever) getRawReport(url string) ([]byte, error) {
+	common := cmd.NewCommonOptions(util.TeamNameSpace(), cmd.NewFactory())
+
+	authSvc, err := common.CreateGitAuthConfigService()
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := buckets.ReadURL(url, timeout, cmd.CreateBucketHTTPFn(authSvc))
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func RetrieveReport(url string) (Report, error) {
+    rawReport, err := retriever.getRawReport(url)
+	if err != nil {
```

Comment:
Use `gofmt` or whatever IDE to make sure you are using TABs consistently.

Reply:
> Use `gofmt` or whatever IDE to make sure you are using TABs consistently.

I'll take care of it. Most of the time I do these things via the make targets. However, the Makefile is currently broken at several places. Hence pr #84

Reply:
> Well the Go compiler will only actually pull in the parts it needs, right?

The rub right now is that the latest jx module file also declares some _replace_ directives. These are not inherited when you then depend on jx. There is an issue between the Kube api and go-client. Until you add these same _replace_ directives you are just going in circles trying to get things to compile. 

Having a more modular apporach, either by restructuring the jx code base to build its binary, but also to have an go gettable libary part or by even pulling out some common functionality into its own libary might help avoiding problems.  

> But yes dealing with `go.mod` is a chore.

Ohh yes.